### ISSUE TITLE
TalkAction.BargeIn should be a bool not a string

### ISSUE THREAD

**Issue by DanielStout5** [Vonage/vonage-dotnet-sdk#211]
BargeIn is either true/false, so there is no reason for this to be anything but a bool property.

Currently it is a string in TalkAction.cs

```
        // Summary:
        //     Set to true so this action is terminated when the user presses a button on the
        //     keypad. Use this feature to enable users to choose an option without having to
        //     listen to the whole message in your Interactive Voice Response (IVR ). If you
        //     set bargeIn to true on one more Stream actions then the next non-stream action
        //     in the NCCO stack must be an input action. The default value is false. Once bargeIn
        //     is set to true it will stay true (even if bargeIn: false is set in a following
        //     action) until an input action is encountered
        [JsonProperty("bargeIn")]
        public string BargeIn
        {
            get;
            set;
        }
```

**Comment by slorello89** [comment]
Hi @DanielStout5 - thank you for reaching out again. You are right, the accepted values of `BargeIn` are `true` and `false`. However, both the boolean and the string should be accepted values by the API so it's not "wrong" for this to be set as a string, as a matter of fact when I was implementing this I saw it allowed both and wanted to leave this flexible. I know it's some weirdness but hopefully, this explains the why behind this.

**Comment by DanielStout5** [comment]
I would call that pretty non-intuitive - if a property can only have values "true" and "false", you'd expect that to be a bool, not a string. I'm glad it works passing as a string, but as an end-user of the library it definitely is strange for a bool property to be typed as string, in my opinion.

**Comment by matt-lethargic** [comment]
This will be changed to a boolean in the next major version of the SDK.
Leaving this issue open as a reminder

**Comment by matt-lethargic** [comment]
This has been addressed in #274 and will be released with v6.x of SDK
